agenda.php

```diff
@@ -702,7 +702,12 @@ public function updateAttendee(string $vCalendarFilename, string $usermail, bool
             // in case of error:  1. update the local agenda, 2. retry once.
             $this->log->warning("Fails to update the event, retrying");
             $this->checkAgenda();
-            $ETag = $this->getEvent($vCalendarFilename)[1]; // retrieve the new ETag (if the event has been updated)
+            $event = $this->getEvent($vCalendarFilename);
+            if(is_null($event)) {
+                $this->log->error("Fails to retrieve the event");
+                return false;
```

Comment:
IIUC returning "false" would now have several posible meaning: it could either mean that the event was deleted from the caldav server (in which case the error is on our side (because our local cache is not up to date) so for instance a retry is not needed) or that a network error occured (in which case a retry could make sense).
To avoid this kind of ambiguity, what about for instance changing this method so that it returns a boolean which say whether a change was made on the caldav server of not (ie: instead of returning NULL), and throwing exceptions in case of errors (so we could naturally have different exceptions for different case of failures, and we could hence let the caller of this method more effectively handles error cases as it wishes (for instance by displaying a more relevant message to the slack user))?

However, that's just a suggestion; it could probably be addressed in a future commit if we want to go in that direction.